src/features/Send/useCases/StartMultiTokenTx/InputReceiver/ResolveAddress.tsx

```diff
@@ -0,0 +1,119 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {Resolution} from '@unstoppabledomains/resolution'
+import _ from 'lodash'
+import React from 'react'
+import {Text, View, ViewProps} from 'react-native'
+import {useQuery, UseQueryOptions} from 'react-query'
+
+import {HelperText} from '../../../../../components'
+import {getNetworkConfigById, NetworkId, normalizeToAddress, YoroiWallet} from '../../../../../yoroi-wallets'
+import {useStrings} from '../../../common/strings'
+import {InputReceiver} from './InputReceiver'
+
+type ReceiverProps = ViewProps & {
+  receiver: string
+  address: string
+  errorMessage: string
+  isLoading: boolean
+  onChangeReceiver: (receiver: string) => void
+}
+export const ResolveAddress = ({
+  isLoading,
+  address,
+  receiver,
+  errorMessage,
+  onChangeReceiver,
+  style,
+  ...props
+}: ReceiverProps) => {
+  const strings = useStrings()
+  const isResolved = !isLoading && isDomain(receiver) && !receiver.includes(address)
+  const isError = errorMessage.length > 0
+
+  return (
+    <View style={style} {...props}>
+      <InputReceiver
+        value={receiver}
+        onChangeText={onChangeReceiver}
+        error={isError}
+        errorText={errorMessage}
+        isLoading={isLoading}
+        autoComplete={false}
+      />
+
+      <HelperText type={isError ? 'error' : 'info'}>
+        {isLoading ? <Text>{strings.pleaseWait}</Text> : <Text>{errorMessage}</Text>}
+      </HelperText>
+
+      {isResolved && (
+        <Text ellipsizeMode="middle" numberOfLines={1}>
+          {`${strings.resolvesTo}: ${address}`}
+        </Text>
+      )}
+    </View>
+  )
+}
+
+export const useReceiver = (
+  {wallet, receiver}: {wallet: YoroiWallet; receiver: string},
+  options?: UseQueryOptions<string, Error, string, [string, 'receiver']>,
+) => {
+  const query = useQuery({
+    queryKey: [receiver, 'receiver'],
```

Comment:
data should come last?
`["receiver", receiver]`

Reply:
I believe since it is cache management the better the cardinality the faster it can check for it, and hence we are not using a complex nested level, it looks more correct to me this way.

Reply:
putting the data first makes invalidating the query impossible

Reply:
I didn't know that was possible. looks pretty ugly, though. I try not to rely on index for anything.

Reply:
it makes sense, done